### PR TITLE
Galaxy 21.09 full release

### PR DESCRIPTION
**Pull request by almahmoud**
This PR complements https://github.com/DataBiosphere/leonardo/pull/2404 for a full release with the sanitizing requirements feature, but still needs some work.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 1]
```diff
@@ -500,6 +493,41 @@ final class LeoAppServiceInterp[F[_]: Parallel](
...
+    implicit as: Ask[F, AppContext]
+  ): F[AppMachineType] =
+    for {
+      ctx <- as.ask
+      _ <- diskSize
+        .traverse_ { size =>
+          if (size.gb < leoKubernetesConfig.galaxyDiskConfig.nfsMinimumDiskSizeGB.gb) {
+            F.raiseError[Unit](
+              BadRequestException(
+                s"Galaxy disk must be at least ${leoKubernetesConfig.galaxyDiskConfig.nfsMinimumDiskSizeGB}",
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:508]: Nit - maybe add `nfsMinimumDiskSizeGB.asString`?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 2]
```diff
@@ -500,6 +493,41 @@ final class LeoAppServiceInterp[F[_]: Parallel](
...
+            )
+          } else F.unit
+        }
+      machineType <- computeService
+        .getMachineType(googleProject,
+                        ZoneName("us-central1-a"),
+                        machineTypeName) //TODO: if use non `us-central1-a` zone for galaxy, this needs to be udpated
+        .flatMap(opt =>
+          F.fromOption(opt,
+                       new LeoException(s"can't find machine config for ${ctx.requestUri}",
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:520]: Maybe this should be `BadRequestException("Unknown machine type ${machineTypeName.asString}")`

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 3]
```diff
@@ -500,6 +493,41 @@ final class LeoAppServiceInterp[F[_]: Parallel](
...
+                                        traceId = Some(ctx.traceId)))
+        )
+      memoryInGb = machineType.getMemoryMb / 1024
+      machineType <- if (memoryInGb < 5)
+        F.raiseError(BadRequestException("Galaxy needs more memorary configuration", Some(ctx.traceId)))
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:525]: Typo - memory

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 4]
```diff
@@ -500,6 +493,41 @@ final class LeoAppServiceInterp[F[_]: Parallel](
...
+        .getMachineType(googleProject,
+                        ZoneName("us-central1-a"),
+                        machineTypeName) //TODO: if use non `us-central1-a` zone for galaxy, this needs to be udpated
+        .flatMap(opt =>
+          F.fromOption(opt,
+                       new LeoException(s"can't find machine config for ${ctx.requestUri}",
+                                        traceId = Some(ctx.traceId)))
+        )
+      memoryInGb = machineType.getMemoryMb / 1024
+      machineType <- if (memoryInGb < 5)
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:524]: Wonder if we should the Galaxy minimum memory/CPU in `reference.conf`, like we have `nfsMinimumDiskSizeGB`? I don't feel too strongly.

**Review by rtitle — APPROVED** [review]
Few comments about messaging, but overall LGTM

**Comment by Qi77Qi** [comment]
@rtitle thanks for reviews! I opened a new PR in https://github.com/DataBiosphere/leonardo/pull/2409 since this one for some reason doesn't trigger automation tests. Think I've addressed all comments in the new PR, lmk if you have other comments over there

**Pull request closed without merging**